packages/i18n-calypso/types/index.d.ts

```diff
@@ -51,17 +51,28 @@ declare namespace i18nCalypso {
 		original: string | { single: string; plural: string; count: number };
 	}
 
-	// Translate hooks force us to open up this type.
-	export type TranslateResult = string | React.ReactFragment;
-
-	export function translate( options: DeprecatedTranslateOptions ): TranslateResult;
-	export function translate( original: string ): TranslateResult;
-	export function translate( original: string, options: TranslateOptions ): TranslateResult;
+	export type TranslateOptionsText = TranslateOptions & { textOnly: true };
+	export type TranslateOptionsPlural = TranslateOptions & { count: number };
+	export type TranslateOptionsPluralText = TranslateOptionsPlural & { textOnly: true };
```

Comment:
Is it possible that this can be defined as `export type TranslateOptionsPluralText = TranslateOptionsPlural & TranslateOptionsText;` or would there be any problems?

Reply:
I would work but I think it's slightly inferior to what I do in the patch. Both `TranslateOptionsPlural` and `TranslateOptionsText` already extend `TranslateOptions` so it would be twice in the result type, like `TO & TO & TT & TP`? It matters very little if at all, but it was the reason for my choice.

Reply:
Got it, thanks for explaining.